mat64/dense.go

```diff
@@ -417,3 +420,25 @@ func (m *Dense) Augment(a, b Matrix) {
 	w := m.View(0, ac, br, bc).(*Dense)
 	w.Copy(b)
 }
+
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+func (m Dense) MarshalBinary() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	enc := binary.NewEncoder(buf)
```

Comment:
We can get away with just row, col and data. There is no need for stride in the serialised data.